packages/dhis-api/src/DhisApi.js

```diff
@@ -132,13 +133,21 @@ export class DhisApi {
   }
 
   async post(endpoint, data, queryParameters) {
+    if (!this.serverPushEnable) {
+      throw new Error(`Attempted to push data to ${this.serverName} that should not be pushed`);
+    }
+
     return this.fetch(endpoint, queryParameters, {
       body: JSON.stringify(data),
       method: 'POST',
     });
   }
 
   async put(endpoint, data, queryParameters) {
+    if (!this.serverPushEnable) {
+      throw new Error(`Attempted to push data to ${this.serverName} that should not be pushed`);
+    }
+
     return this.fetch(endpoint, queryParameters, {
       body: JSON.stringify(data),
       method: 'PUT',
```

Comment:
Can we add it to ` delete()` as well. I think that's all of them that mutate data